Build fails on JDK10 due to the presence of "finalize" methods

**Problem description**

Since JDK10 'finalize' has been deprecated, so while building on JDK10 the build fails

Errors are like this one:
```
Task :client:compileJava 
/xxxxx/pravega/client/src/main/java/io/pravega/client/netty/impl/ConnectionFactoryImpl.java:225: warning: [deprecation] finalize() in Object has been deprecated
    protected void finalize() {


```
**Problem location**
Source code                       

**Suggestions for an improvement**
Drop "finalize" where it is not useful, add SuppressWarning annotation for the rest (and create issues for refactors)

--- a/client/src/main/java/io/pravega/client/netty/impl/ConnectionFactoryImpl.java
+++ b/client/src/main/java/io/pravega/client/netty/impl/ConnectionFactoryImpl.java
@@ -221,8 +221,4 @@ public final class ConnectionFactoryImpl implements ConnectionFactory {
         return channelGroup.stream().filter(Channel::isActive).collect(Collectors.toList());
     }
 
-    @Override
-    protected void finalize() {
-        close();
-    }
 }

--- a/shared/metrics/src/main/java/io/pravega/shared/metrics/OpStatsLoggerImpl.java
+++ b/shared/metrics/src/main/java/io/pravega/shared/metrics/OpStatsLoggerImpl.java
@@ -50,10 +50,6 @@ class OpStatsLoggerImpl implements OpStatsLogger {
         this.metricRegistry.remove(this.failName);
     }
 
-    @Override
-    protected void finalize() {
-        close();
-    }
 
     //endregion
 

--- a/shared/metrics/src/main/java/io/pravega/shared/metrics/StatsLoggerImpl.java
+++ b/shared/metrics/src/main/java/io/pravega/shared/metrics/StatsLoggerImpl.java
@@ -95,12 +95,7 @@ public class StatsLoggerImpl implements StatsLogger {
         @Override
         public void close() {
             metrics.remove(this.name);
-        }
-
-        @Override
-        protected void finalize() {
-            close();
-        }
+        }        
 
         @Override
         public synchronized void clear() {
@@ -146,10 +141,6 @@ public class StatsLoggerImpl implements StatsLogger {
             metrics.remove(this.name);
         }
 
-        @Override
-        protected void finalize() {
-            close();
-        }
     }
 
     private class MeterImpl implements Meter {
@@ -165,11 +156,6 @@ public class StatsLoggerImpl implements StatsLogger {
         @Override
         public void close() {
             metrics.remove(this.name);
-        }
-
-        @Override
-        protected void finalize() {
-            close();
         }
 
         @Override

